### PR TITLE
Fix/update/javascript/faq/page/handling/issue#2376

### LINKED ISSUE #2376: Update Javascript handling the FAQ page
#### Description
Some of the legacy Javascript can now be updated so we can eventually remove the depedency on jquery.

As part of this we need to update [this file](https://github.com/TheOdinProject/theodinproject/blob/main/app/javascript/src/js/faq.js) to use a Stimulus Controller.

Suggest reading to help you: https://stimulus.hotwired.dev/ and https://www.betterstimulus.com/

#### Acceptance Criteria:

1. The dependency on JQuery is removed
2. The FAQ page operates as it currently does
3. A stimulus controller is created to handle the interaction

#### Additional Information:

Stimulus controllers are all about finding the correct level of abstraction. The suggested reading for better stimulus covers this.

### PR DESCRIPTION
**Pull request by linkonsat**
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behaviour
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes

Complete the following checkbox ONLY IF it is applicable to your PR. You can complete it later if it is not currently applicable:
-   [x] I have verified all tests and linters pass against my changes, and/or I have included automated tests where applicable

<hr>

**1. Because:**
This PR is the updated version of an already open one due to the local repo being deleted by accident. Incorporates the changes requested [here](https://github.com/TheOdinProject/theodinproject/pull/2485). As well as resolves an open issue.

Closes #2376 

**2. This PR:**

**3. Additional Information:**

### CODE REVIEW

File: app/javascript/controllers/reveal_controller.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// reveal_controller.js
```
Comment (KevinMulhern) [inline app/javascript/controllers/reveal_controller.js:1]: Sorry for the confusion @linkonsat, I know you got these comments from the example code, I should have made it clear the comments were annotations. The code should be clear enough without them though. Can we remove the comments in the js and HTML file please?

File: app/views/static_pages/faq/_faq_card.html.erb [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
   </a>
 
-  <div id="faq-item-<%= index %>" class="collapse" role="tabpanel" aria-labelledby="faq-item-<%= index %>">
+  <div class="hidden" data-reveal-target="content"> <!-- content target -->
     <div class="card-block">
       <%= faq_item[:answer].html_safe %>
     </div>
-  </div> <!-- /.collapse -->
-</div> <!-- /.faq-card -->
+  </div></div> <!-- /.faq-card -->
```
Comment (KevinMulhern) [inline app/views/static_pages/faq/_faq_card.html.erb:15]: Can we put this closing div on a new line please?

File: app/views/static_pages/faq/_faq_card.html.erb [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
+<div class="card faq-card" data-controller="reveal"> <!-- controller at top level -->
+  <a role="button" data-action="click->reveal#toggle"> <!-- toggle action to show and hide the content -->
     <div class="card-header faq-card-header" role="tab">
       <h2 class="secondary bold faq-question-title text-2xl">
         <%= faq_item[:question] %>
       </h2>
-      <i class="fas fa-plus camel expander"></i>
-      <i class="fas fa-minus camel minimizer"></i>
-    </div> <!-- /.card-header-->
+      <i class="fas fa-chevron-down"></i>
```
Comment (KevinMulhern) [inline app/views/static_pages/faq/_faq_card.html.erb:7]: Just a little polish to make the chevrons fit better with the text sizing and color on the faq item:
```suggestion
      <i class="fas fa-chevron-down text-xl text-gray-500"></i>
```

**Review by KevinMulhern — CHANGES_REQUESTED** [review]

**Comment by linkonsat** [comment]
Changes were incorporated! I didn't see anything else that seemed off to me styling wise but if there is anything I can change let me know.

File: app/javascript/controllers/reveal_controller.js (hunk 1 above)
Comment (KevinMulhern) [inline app/javascript/controllers/reveal_controller.js:1]: Can we remove the `// reveal_controller.js` comment as well please @linkonsat ?

File: app/views/static_pages/faq/_faq_card.html.erb [hunk 4]
```diff
@@ -1,17 +1,16 @@
...
 
-  <div id="faq-item-<%= index %>" class="collapse" role="tabpanel" aria-labelledby="faq-item-<%= index %>">
+  <div class="hidden" data-reveal-target="content">
     <div class="card-block">
       <%= faq_item[:answer].html_safe %>
     </div>
-  </div> <!-- /.collapse -->
-</div> <!-- /.faq-card -->
+  </div>
+  </div>
```
Comment (KevinMulhern) [inline app/views/static_pages/faq/_faq_card.html.erb:16]: Looks like this just needs de-dented by one so it's flush with the opening tag.

**Comment by linkonsat** [comment]
Changes incorporated. Do you want me to squash some of the commits together since it's quite a few? Thanks for your patience as well Kevin. Doing this PR definitely helped uncover some unclean code habits I had so appreciate the feedback a lot! Definitely going to incorporate the feedback in future PR's.

**Comment by KevinMulhern** [comment]
Thanks for making those changes @linkonsat, this is looking great, nice work!

Yeah, lets squash the commits and get this merged mate 💪

**Comment by linkonsat** [comment]
Squashed together the commits but for some reason, it pushed a whole bunch of other ones despite not having them in the rebase. Practiced squashing a branch in one of my repositories and didn't get the same behavior so opened up a new PR without all the other commits [here](https://github.com/TheOdinProject/theodinproject/pull/2719).

**Pull request closed without merging**